geode-core/src/main/java/org/apache/geode/cache/query/internal/DefaultQuery.java

```diff
@@ -477,7 +440,7 @@ public Object executeUsingContext(ExecutionContext context) throws FunctionDomai
       updateStatistics(endTime - startTime);
       pdxClassToFieldsMap.remove();
       pdxClassToMethodsMap.remove();
-      queryCanceled.remove();
+      ExecutionContext.isCanceled.remove();
```

Comment:
Giving `DefaultQuery` responsibility for removing `ExecutionContext`'s thread local, breaks encapsulation.

One solution that occurs to me is to put `ExecutionContext` in charge of the scope of work, via a method ref. Maybe that's just 💋 on a :pig: tho. The real answer is to get rid of the thread local in the next refactor. 

Never mind :)

Reply:
Yeah I had the same internal conflict.  This actually isn't even the best place to remove this thread local even when it was on the default query.  It really should be in the finalizer after we stop monitoring.  But I think its a fight for another day.